app/registration/controllers/patientCommonController.js

```diff
@@ -81,40 +95,33 @@
 
 
     function getDeathConcepts() {
-      var deathConcept;
-      var deathConceptValue;
-      $http({
-        url: '/openmrs/ws/rest/v1/systemsetting',
-        method: 'GET',
-        params: {
-          q: 'concept.causeOfDeath',
-          v: 'full'
-        },
-        withCredentials: true,
-        transformResponse: [function (data) {
-          deathConcept = JSON.parse(data);
-          deathConceptValue = deathConcept.results[0].value;
-          $http.get(Bahmni.Common.Constants.conceptUrl, {
-            params: {
-              q: deathConceptValue,
-              v: 'custom:(uuid,name,set,answers:(uuid,display,name:(uuid,name),retired))'
-            },
-            withCredentials: true
-          }).then(function (results) {
-            vm.deathConcepts = results.data.results[0] !== null ? results.data.results[0].answers : [];
-            vm.deathConcepts = filterRetireDeathConcepts(vm.deathConcepts);
-          });
-        }]
-      });
+      conceptService.getDeathConcepts()
```

Comment:
This promise needs to be returned otherwise it brakes the chain in activate function.